packages/jest-snapshot/src/__tests__/printDiffOrStringified.test.ts

```diff
@@ -0,0 +1,318 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import ansiRegex from 'ansi-regex';
+import style from 'ansi-styles';
+import {printDiffOrStringified} from '../print';
+import {serialize, unescape} from '../utils';
+
+// This is an experiment to read snapshots more easily:
+// * to avoid first line misaligned because of opening double quote mark,
```

Comment:
Nice! Does some things more and some things less than https://github.com/ikatyang/jest-snapshot-serializer-raw - maybe we should think about making a full-featured serializer for edge case strings 😄

Reply:
Thank you for the link! It answers questions in back of my mind about prettier snapshots.

So I am exploring if this pull request can also highlight substrings when:

* received is string
* expected is serialized by **application-specific plugin** instead of default stringify